### PR TITLE
Provide System.DirectoryServices.Protocols package readme

### PR DESCRIPTION
**Pull request by eNeRGy164**
Fill out details for the `System.DirectoryServices.Protocols` NuGet package readme.

See #92228

### CODE REVIEW

**Comment by danmoseley** [comment]
Thanks @eNeRGy164

File: src/libraries/System.DirectoryServices.Protocols/src/PACKAGE.md [hunk 1]
```diff
@@ -2,43 +2,69 @@
...
+
+// Some credentials
+connection.Credential = new NetworkCredential(dn, password);
+
+// Connect to the server
+connection.Bind();
+
+// Perform LDAP operations
+
+connection.Dispose();
```
[the remark this answers is not in the document]
Comment (eNeRGy164) [reply src/libraries/System.DirectoryServices.Protocols/src/PACKAGE.md:37]: I was doubting about that, this makes it quite explicit that you should dispose, and I think the new way of `using` without `{}` is not recognizable enough.

File: src/libraries/System.DirectoryServices.Protocols/src/PACKAGE.md [hunk 2]
```diff
@@ -2,43 +2,69 @@
 
 <!-- A description of the package and where one can find more documentation -->
 
+System.DirectoryServices.Protocols provides a managed implementation of Lightweight Directory Access Protocol (LDAP) version 3 and Directory Services Markup Language (DSML) version 2.0 (V2) standards.
 
+It primarily utilizes the `LdapConnection` type for interacting with LDAP servers, utilizing system native libraries to establish TCP/IP or UDP LDAP connections.
+Initially supporting only Windows, the assembly now also supports Unix environments, although not yet achieving feature parity between the two.
```
[the remark this answers is not in the document]
Comment (eNeRGy164) [reply src/libraries/System.DirectoryServices.Protocols/src/PACKAGE.md:8]: I have to rely on the source code and documentation, and they are quite sparse. I found some comments on client and server cert options, added that as suggestion.

**Review by danmoseley — APPROVED** [review]
Looks good to me but owner of the area should sign off (and confirm code sample)

**Comment by eNeRGy164** [comment]
> Feel like doing more of the package readmes? Now you got warmed up 😀

#93485, #93489, #93516 😝

**Comment by danmoseley** [comment]
Oh, whoops 🙂

**Comment by ViktorHofer** [comment]
@dotnet/area-system-directoryservices please review this package readme PR before EOW so that we can get the change still ported into .NET 8.

File: src/libraries/System.DirectoryServices.Protocols/src/PACKAGE.md [hunk 3]
```diff
@@ -2,43 +2,68 @@
...
 <!-- The main types provided in this library -->
 
 The main types provided by this library are:
 
-* ``
-* ``
-* ``
+* `LdapConnection`
+* `DirectoryAttribute`
+* `DirectoryEntry`
```
Comment (buyaa-n) [inline src/libraries/System.DirectoryServices.Protocols/src/PACKAGE.md:47]: Wonder why `DirectoryEntry` added, don't see this type in the `System.DirectoryServices.Protocols` namespace, don't see any API that returns `System.DirectoryServices.DirectoryEntry`

**Review by buyaa-n — APPROVED** [review]
Thank you @eNeRGy164, left a comment, overall LGTM

File: src/libraries/System.DirectoryServices.Protocols/src/PACKAGE.md [hunk 4]
```diff
@@ -2,43 +2,68 @@
...
+* `DirectoryOperation`
+* `DirectoryRequest`
+* `DirectoryResponse`
```
Comment (ViktorHofer) [inline src/libraries/System.DirectoryServices.Protocols/src/PACKAGE.md:50]: Same comment as someone else provide in another one of your PRs, please use the type's fully qualified name.

File: src/libraries/System.DirectoryServices.Protocols/src/PACKAGE.md [hunk 5]
```diff
@@ -2,43 +2,68 @@
...
 
 ## Additional Documentation
 
 <!-- Links to further documentation. Remove conceptual documentation if not available for the library. -->
 
-* [Conceptual documentation](https://learn.microsoft.com/en-us/dotnet/standard/serialization/**LIBRARYNAME**/overview)
-* [API documentation](https://learn.microsoft.com/en-us/dotnet/api/**LIBRARYNAME**)
+* [API documentation](https://learn.microsoft.com/en-us/dotnet/api/system.directoryservices.protocols)
+* [Active Directory Domain Services](https://learn.microsoft.com/windows/win32/ad/active-directory-domain-services)
```
Comment (ViktorHofer) [inline src/libraries/System.DirectoryServices.Protocols/src/PACKAGE.md:57]: Also same as someone else already suggested in another of your PRs: remove the "en-us" part of the links to make this culture agnostic.

**Comment by buyaa-n** [comment]
The failure unrelated and doesn't seem actionable.

**Pull request merged**